desc/objectives/utils.py

```diff
@@ -130,32 +149,57 @@ def factorize_linear_constraints(objective, constraint):  # noqa: C901
             )
         A = A[unfixed_rows][:, unfixed_idx]
         b = b[unfixed_rows]
+
     unfixed_idx = indices_idx
+    fixed_idx = np.delete(np.arange(xp.size), unfixed_idx)
+
+    # compute x_scale if not provided
+    if x_scale == "auto":
+        if things is None:
+            things = objective.things
+        else:
+            things = [things] if not isinstance(things, list) else things
+            things = [get_instance(things, type(t)) for t in objective.things]
+        x_scale = objective.x(*things)
+    errorif(
+        x_scale.shape != xp.shape,
+        ValueError,
+        "x_scale must be the same size as the full state vector.",
+    )
+    D = np.where(np.abs(x_scale) < 2e1, 1, np.abs(x_scale))
```

Comment:
something more like 100-1000 might be better to clearly separate stuff like R, Z (which could be ~20 for reactor size stuff) from pressure, current, etc which would usually be > 1e3~1e4